### PR TITLE
Add Stack::exists?

### PR DESCRIPTION
**Pull request by jpslav**
Add Stack::exists?

### CODE REVIEW

File: lib/openstax/aws/stack.rb [hunk 1]
```diff
@@ -297,7 +297,16 @@ def self.format_hash_as_stack_parameters(params={})
     end
 
     def status
-      aws_stack.stack_status
+      begin
+        aws_stack.stack_status
+      rescue Aws::CloudFormation::Errors::ValidationError => ee
```
Comment (steveburkett) [inline lib/openstax/aws/stack.rb]: minor.  https://github.com/rubocop-hq/ruby-style-guide shows rescue using a shorthand convention of 'e' over 'ee'

File: lib/openstax/aws/stack.rb [hunk 2]
```diff
@@ -297,7 +297,16 @@ def self.format_hash_as_stack_parameters(params={})
...
+        case ee.message
+        when /Stack.*does not exist/
+          "DOES_NOT_EXIST"
+        else
```
Comment (steveburkett) [inline lib/openstax/aws/stack.rb]: "DOES_NOT_EXIST" as a class constant? (then we can have just one copy of this string, referred to in two different places, and potentially more down the road.)

File: lib/openstax/aws/stack.rb [hunk 3]
```diff
@@ -319,6 +328,10 @@ def deleting?
       "DELETE_IN_PROGRESS" == status
     end
 
+    def exists?
```
Comment (steveburkett) [inline lib/openstax/aws/stack.rb]: a (simple) unit test for this?

File: lib/openstax/aws/stack.rb (hunk 1 above)
Comment (jpslav) [reply lib/openstax/aws/stack.rb]: I can't do single letter variable names :-)

File: lib/openstax/aws/stack.rb (hunk 3 above)
Comment (jpslav) [reply lib/openstax/aws/stack.rb]: Yeah sorry didn't mean for you to review this until I had written the test.

**Pull request merged**